src/conduits/RwaMultiSwapOutputConduit.sol

```diff
@@ -0,0 +1,463 @@
+        _doQuit(wad);
+    }
+
+    /**
+     * @notice Flushes out `amt` of `token` sitting in this contract to `usr` address.
+     * @dev Can only be called by the admin.
+     * @param token Token address.
+     * @param usr Destination address.
+     * @param amt Token amount.
+     */
+    function yank(
+        address token,
+        address usr,
+        uint256 amt
+    ) external auth {
+        GemAbstract(token).transfer(usr, amt);
+        emit Yank(token, usr, amt);
+    }
+
+    /**
+     * @notice Return Gem address of the selected PSM
+     * @return gem Gem address.
+     */
+    function gem() external view returns (address) {
+        return psm != address(0) ? GemJoinAbstract(PsmAbstract(psm).gemJoin()).gem() : address(0);
+    }
+
+    /**
+     * @notice Calculates the amount of GEM received for swapping `wad` of DAI.
+     * @param wad DAI amount.
+     * @return amt Expected GEM amount.
+     */
+    function expectedGemAmt(uint256 wad) public view returns (uint256 amt) {
+        if (psm == address(0)) return 0;
+
+        uint256 decimals = GemAbstract(GemJoinAbstract(PsmAbstract(psm).gemJoin()).gem()).decimals();
+        // If `psm` set, its gem is guaranteed to have 18 decimals or less.
+        uint256 to18ConversionFactor = 10**(18 - decimals);
+        return _mul(wad, WAD) / _mul(_add(WAD, PsmAbstract(psm).tout()), to18ConversionFactor);
+    }
+
+    /**
+     * @notice Calculates the required amount of DAI to get `amt` amount of GEM.
+     * @param amt GEM amount.
+     * @return wad Required DAI amount.
+     */
+    function requiredDaiWad(uint256 amt) external view returns (uint256 wad) {
+        if (psm == address(0)) return 0;
+
+        uint256 decimals = GemAbstract(GemJoinAbstract(PsmAbstract(psm).gemJoin()).gem()).decimals();
+        // If `psm` set, its gem is guaranteed to have 18 decimals or less.
+        uint256 amt18 = _mul(amt, 10**(18 - decimals));
+        uint256 fee = _mul(amt18, PsmAbstract(psm).tout()) / WAD;
+        return _add(amt18, fee);
+    }
+
+    /**
+     * @notice Swaps the specified amount of DAI into GEM through the PSM and push it to the recipient address.
+     * @param wad DAI amount.
+     */
+    function _doPush(uint256 wad) internal {
+        require(to != address(0), "RwaMultiSwapOutputConduit/to-not-picked");
+        require(psm != address(0), "RwaMultiSwapOutputConduit/psm-not-hooked");
+
+        // We might lose some dust here because of rounding errors. I.e.: USDC has 6 dec and DAI has 18.
+        uint256 gemAmt = expectedGemAmt(wad);
+        require(gemAmt > 0, "RwaMultiSwapOutputConduit/insufficient-swap-gem-amount");
+
+        address recipient = to;
+        address _psm = psm;
+        address _gem = GemJoinAbstract(PsmAbstract(psm).gemJoin()).gem();
+        to = address(0);
+        psm = address(0);
+
+        PsmAbstract(_psm).buyGem(recipient, gemAmt);
+        emit Push(_psm, _gem, recipient, gemAmt);
```

Comment:
There's no naming collision for `gem`, so maybe we should drop the `_` prefix:

```suggestion
        address gem = GemJoinAbstract(PsmAbstract(psm).gemJoin()).gem();
        to = address(0);
        psm = address(0);

        PsmAbstract(_psm).buyGem(recipient, gemAmt);
        emit Push(_psm, gem, recipient, gemAmt);
```

Reply:
Actually, the name sort of collides with the getter `gem()` function. Even though there is no shadowing at code level, maybe it's a better idea to have a separate name.